make test-tools-image fails on Apple silicon

<!-- If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack). -->

Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [ ] I've pasted the output of `argocd version`.

**Describe the bug**

<!-- A clear and concise description of what the bug is. -->

On any M1 Mac, the `test-tools-image` make target fails with the following error:

```
 => ERROR [stage-5 13/21] COPY --from=redis /usr/lib/x86_64-linux-gnu/libssl.so.1.1 /usr/lib/x86_64-linux-gnu/                         0.0s
 => ERROR [stage-5 14/21] COPY --from=redis /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1 /usr/lib/x86_64-linux-gnu/                      0.0s
------
 > [stage-5 13/21] COPY --from=redis /usr/lib/x86_64-linux-gnu/libssl.so.1.1 /usr/lib/x86_64-linux-gnu/:
------
------
 > [stage-5 14/21] COPY --from=redis /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1 /usr/lib/x86_64-linux-gnu/:
------
failed to compute cache key: "/usr/lib/x86_64-linux-gnu/libcrypto.so.1.1" not found: not found
make: *** [test-tools-image] Error 1
```

The issue is due to lib paths being different from one CPU architecture to the next.

**To Reproduce**

<!-- A list of the steps required to reproduce the issue. Best of all, give us the URL to a repository that exhibits this issue. -->

Run `make test-tools-image` (or any target that depends on that one) on an M1 Mac.

**Expected behavior**

<!-- A clear and concise description of what you expected to happen. -->

The target should work equally well for all CPU architectures; not just amd64.

--- a/test/container/Dockerfile
+++ b/test/container/Dockerfile
@@ -1,5 +1,10 @@
 FROM docker.io/library/redis:7.0.0 as redis
 
+# There are libraries we will want to copy from here in the final stage of the
+# build, but the COPY directive does not have a way to determine system
+# architecture, so we create a symlink here to facilitate copying.
+RUN ln -s /usr/lib/$(uname -m)-linux-gnu /usr/lib/linux-gnu 
+
 FROM docker.io/library/node:12.18.4-buster as node
 
 FROM docker.io/library/golang:1.18 as golang
@@ -66,8 +71,11 @@ COPY --from=redis /usr/local/bin/* /usr/local/bin/
 
 # Copy redis dependencies/shared libraries
 # Ubuntu 22.04+ has moved to OpenSSL3 and no longer provides these libraries
-COPY --from=redis /usr/lib/x86_64-linux-gnu/libssl.so.1.1 /usr/lib/x86_64-linux-gnu/
-COPY --from=redis /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1 /usr/lib/x86_64-linux-gnu/
+COPY --from=redis /usr/lib/linux-gnu/libssl.so.1.1 /usr/lib/linux-gnu/
+COPY --from=redis /usr/lib/linux-gnu/libcrypto.so.1.1 /usr/lib/linux-gnu/
+RUN mv /usr/lib/linux-gnu/libssl.so.1.1 /usr/lib/$(uname -m)-linux-gnu/ && \
+    mv /usr/lib/linux-gnu/libcrypto.so.1.1 /usr/lib/$(uname -m)-linux-gnu/ && \
+    rm -rf /usr/lib/linux-gnu/
 
 # Copy registry binaries to the image
 COPY --from=registry /bin/registry /usr/local/bin/

